src/arc_utilities/ros_helpers.py

```diff
@@ -29,10 +29,67 @@ def callback(self, msg):
         with self.lock:
             self.data = msg
 
-    def get(self):
+    def get(self, block_on_none=False):
         """
         Returns the latest msg from the subscribed topic
+
+        Parameters:
+        block_on_none (bool): block return until data has been received
         """
+        wait_for(lambda: not (block_on_none and self.data is None))
+            
         with self.lock:
             return self.data
+
     
+def wait_for(func):
+    """
+    Waits for function evaluation to be true. Exits cleanly from ros.
+
+    Introduces sleep delay, not recommended for time critical operations
+    """
+    
+    while not func() and not rospy.is_shutdown():
+        time.sleep(0.01)
+
+def joy_to_xbox(joy):
+    """
+    Transforms a joystick sensor_msg to a XBox controller for easier code readability
+    
+    Parameters:
+    joy (sensor_msgs/Joy): xbox msg
+
+    Returns:
+    xbox struct where fields are the button names
+    """
+    class Xbox_msg():
+        pass
+    x = Xbox_msg()
+    x.A, x.B, x.X, x.Y, x.LB, x.RB, \
+        x.back, x.start, x.power,\
+        x.stick_button_left, x.stick_button_right, \
+        x.DL, x.DR, x.DU, x.DD = joy.buttons
+    x.LH, x.LV, x.LT, x.RH, x.RV, x.RT, x.DH, x.DV = joy.axes
+    return x
+        
+
+
+class Xbox():
+    def __init__(self):
+        self.xbox_listener = Listener("/joy", Joy)
```

Comment:
I don't like this hardcoded global namespace here, is there any reason not to make it relative at least?

Reply:
You mean the "/joy" topic?
The "joy_node" from the ros package "joy" publishes on topic "/joy". 
If we get a second joystick we'll want to make some code changes anyway.

We could do:
``` def __init__(self, joy_topic = "/joy"):```

Reply:
I'm asking about `"/joy"` vs `"joy"`. Does the joy node publish to a global namespace or a relative one? I'd prefer that the default be topic (wherever we encode it) be a relative topic name rather than a global one unless there is a strong reason to use a global topic name.

Reply:
Why do you prefer a relative namespace? 

Currently the joystick node will publish in "/joy".
I may want to run my node in "/brad_test"
If the listener used relative namespaces, it'd be listening to the meaningless "/brad_test/joy" topic.

I imagine you have a different use case in mind?